### PR TITLE
Add a test to ensure all rst ini blocks parse correctly

### LINKED ISSUE #900: ensure tox.ini examples in our documentation can be parsed
Would be nice if we could ensure that our examples in our documentation parse correctly. 

@asottile @nicoddemus how hard would it be to generate for pytest extra tests by parsing the ``rst`` files inside the documentation, looking for ``.. codeblock:: ini`` and wrapping it into a generated test (e.g. ``parse_config("ini")``). I assume this requires adding an additional test discovery mechanism and an additional test runner for it. Does anyone have examples of similar? Docstrings are sort of similar to this.

### PR DESCRIPTION
**Pull request by asottile**
Resolves #900

### CODE REVIEW

File: tests/test_docs.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    r"((?P=indent) +:.*\n)*"
+    r"\n*"
+    r")"
+    r"(?P<code>(^((?P=indent) +.*)?\n)+)",
+    re.MULTILINE,
+)
+
+
+rst_files = []
+for root, _, filenames in os.walk("doc"):
```
Comment (gaborbernat) [inline tests/test_docs.py]: does pytest guarantee the current working directory is the root; or should we do some kind of relative from this file logic here?

Comment (asottile) [reply tests/test_docs.py]: it doesn't if some other test runs `chdir` then this points at the wrong file locations -- I can `abspath` + join with `__file__` to make this better

Comment (gaborbernat) [reply tests/test_docs.py]: Yeah I think that would make this more robust; otherwise looks good :+1:

Comment (gaborbernat) [reply tests/test_docs.py]: Also in the CI I see:

```
SKIP [1] tests/test_docs.py:27: got empty parameter set ['filename'], function test_all_rst_ini_blocks_parse at /home/travis/build/tox-dev/tox/tests/test_docs.py:26
```

Comment (asottile) [reply tests/test_docs.py]: heh, probably due to the `changedir`

**Comment by obestwalter** [comment]
Nice one :)

**Comment by nicoddemus** [comment]
👍 simple and to the point. 😁

**Comment by asottile** [comment]
I suspect the tests are failing (and unreliably) due to random test ordering -- I'll see if I can't improve that...

**Comment by asottile** [comment]
hah wow, this wasn't actually testing what I expected -- turns out `parseconfig` will just "try its best" when fed garbage.

**Comment by asottile** [comment]
want to re-review now that I've rewritten a bunch of `.rst`?

**Pull request merged**